### PR TITLE
Fix controlled input values out of sync with dom

### LINKED ISSUE #1899: Uncontrolled input
Same logic with different libraries works different.

React - ✅
https://codesandbox.io/s/react-controlled-input-egxgu

Preact - ❌
https://codesandbox.io/s/preact-controlled-input-kib3p

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Fixes #1899 .
Adds `+15 B`

### CODE REVIEW

File: src/diff/props.js [hunk 1]
```diff
@@ -122,5 +122,6 @@ function setProperty(dom, name, value, oldValue, isSvg) {
  * @private
  */
 function eventProxy(e) {
-	return this._listeners[e.type](options.event ? options.event(e) : e);
+	this._listeners[e.type](options.event ? options.event(e) : e);
```
Comment (andrewiggins) [inline src/diff/props.js]: Do we need to still capture the return value from the handler and return it?

File: src/diff/props.js [hunk 2]
```diff
@@ -122,5 +122,6 @@ function setProperty(dom, name, value, oldValue, isSvg) {
...
+	if (this._lastValue!=this.value) this.value = this._lastValue;
```
Comment (andrewiggins) [inline src/diff/props.js]: This logic will apply for any event added to an input element, right? So if I add `onFocus`, I'll need to also make it a controlled input in order for the input to work? Just checking if I understand how this works.

File: src/diff/index.js [hunk 3]
```diff
@@ -255,6 +255,10 @@ function diffElementNodes(dom, newVNode, oldVNode, context, isSvg, excessDomChil
 		// (as above, don't diff props during hydration)
 		if (!isHydrating) {
 			if (('value' in newProps) && newProps.value!==undefined && newProps.value !== dom.value) dom.value = newProps.value==null ? '' : newProps.value;
+			// preact/#1899
+			// We need this value for input masking.
+			dom._lastValue = dom.value;
```
Comment (developit) [inline src/diff/index.js]: should we add a `mangle.json` mapping for this? We could reuse a property name that's only used on component or vnode objects right now, like `__s` ("next state").

File: src/diff/props.js (hunk 2 above)
Comment (developit) [reply src/diff/props.js]: ahhhh good catch. Maybe this should guard and only apply for `onChange`/`onInput`? Though right now it means this works with either, and guards against state mutations in focus/blur.

File: src/diff/props.js (hunk 1 above)
Comment (marvinhagemeister) [reply src/diff/props.js]: I don't think we do. We attach `eventProxy` like a standard event handler and those don't return anything:

```
dom.addEventListener("input"; e => {
  // No need to return anything here
});
```

**Comment by marvinhagemeister** [comment]
I'll leave this open as I want to check if this PR leads to cursor jumps or something.

File: src/diff/index.js (hunk 3 above)
Comment (marvinhagemeister) [reply src/diff/index.js]: Done :ballot_box_with_check:

**Comment by marvinhagemeister** [comment]
Just checked, this solution leads to cursor jumping :cry:

**Pull request closed without merging**